docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,646 @@ under the License.
-execution. The lifetime of this Flink Cluster is not bound to the lifetime of any Flink Job.
-Formerly, a Flink Session Cluster was also known as a Flink Cluster in *session mode*. Compare to
-[Flink Application Cluster](#flink-application-cluster).
+This is a technique where state data is spilled to disk before JVM heap memory is exhausted.
 
 #### State Backend
 
-For stream processing programs, the State Backend of a [Flink Job](#flink-job) determines how its
-[state](#managed-state) is stored on each TaskManager (Java Heap of TaskManager or (embedded)
-RocksDB).
+For stream processing programs, the state backend of a [Flink job](#(flink)-job) determines how its
+[state](#managed-state) is stored on each [TaskManager](#taskmanager).
 
-#### Sub-Task
+Two implementations of state backends are available. One is based on RocksDB, an embedded key/value 
+store that keeps its working state on disk, and the other is heap-based that keeps its working state 
+in memory, on the Java heap.
 
-A Sub-Task is a [Task](#task) responsible for processing a [partition](#partition) of
-the data stream. The term "Sub-Task" emphasizes that there are multiple parallel Tasks for the same
-[Operator](#operator) or [Operator Chain](#operator-chain).
+#### Stream
+
+This is a concept describing a continuous flow of data (generated by various sources) that is fed into 
+stream processing systems to be processed and analyzed in real time. It consists of a series of data 
+elements ordered in time. 
+
+[DataStream](#flink-datastream) is the representation of this concept inside Flink's DataStream API 
+and Table API.
+
+#### Stream Barriers
+
+A core element of Flink's distributed snapshotting. Stream barriers are injected into the [DataStream](#datastream) 
+and flow with the [records](#record) as part of the DataStream. Barriers never overtake records and 
+flow strictly in line. A barrier separates the records in the DataStream into the set of records that 
+goes into the current snapshot, and the records that go into the next snapshot.
+
+#### Subtask
+
+A subtask is an abstract concept that describes a [task](#task) responsible for processing a [partition](#partition) 
+of the [DataStream](#datastream). The term "subtask" emphasizes that there are multiple parallel tasks 
+for the same [operator](#operator) or [operator chain](#operator-chain).
 
 #### Table Program
 
 A generic term for pipelines declared with Flink's relational APIs (Table API or SQL).
 
 #### Task
 
-Node of a [Physical Graph](#physical-graph). A task is the basic unit of work, which is executed by
-Flink's runtime. Tasks encapsulate exactly one parallel instance of an
-[Operator](#operator) or [Operator Chain](#operator-chain).
+This is a node in a [physical graph](#physical-graph). A task is the basic unit of work which is executed 
+by Flink's runtime. Tasks encapsulate exactly one parallel instance of an [operator](#operator) or
+[operator chain](#operator-chain).
+
+#### Task Chaining
```

Comment:
Tasks aren't chained, in my opinion. Normally this is called _operator chaining_.